src/trackEventMethodDecorator.js

```diff
@@ -22,10 +22,15 @@ export default function trackEventMethodDecorator(trackingData = {}) {
 
         const fn = Reflect.apply(decoratedFn, this, args);
         if (Promise && Promise.resolve(fn) === fn) {
-          return fn.then(trackEvent.bind(this)).catch(error => {
-            trackEvent(null, error);
-            throw error;
-          });
+          return fn
+            .then(trackEvent.bind(this))
+            .then(() => {
+              return fn;
+            })
```

Comment:
nit: mind rewriting as:

```js
.then(() => fn)
```

Reply:
Fixed with d7adf1f
I also added an e2e test as suggested. 

When writing the test, I noticed that the tracking function breaks if a promise is rejected. It will fail on  `{ value }` since value is set to null after reject. So when using the example in the readme:

` @track((props, state, methodArgs, [{ value }, err]) => { }` <-- will give TypeError

`           .catch(error => {
              trackEvent(null, error);
              throw error;
            });` <-- trackEventMethodDecorators.js line 28, value is set to null

So the happy flow is working now, but maybe this should be fixed as well. (changing null to {}?)

Reply:
Yea that makes sense, good catch! Mind making that change as a separate PR (will need an accompanying test update)